api/v1beta1/azurecluster_validation.go

```diff
@@ -611,3 +619,84 @@ func validateServiceEndpointLocationName(location string, fldPath *field.Path) *
 	}
 	return nil
 }
+
+func validatePrivateEndpoints(privateEndpointSpecs []PrivateEndpointSpec, subnetCIDRs []string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	privateEndpoints := make(map[string]struct{}, len(privateEndpointSpecs))
+
+	for i, pe := range privateEndpointSpecs {
+		if err := validatePrivateEndpointName(pe.Name, fldPath.Index(i).Child("name")); err != nil {
+			allErrs = append(allErrs, err)
+		}
+		if _, ok := privateEndpoints[pe.Name]; ok {
+			allErrs = append(allErrs, field.Duplicate(fldPath.Index(i).Child("name"), pe.Name))
```

Comment:
When I comment this out and try providing duplicate names here, I still see an error so I think it's safe to remove the logic to track names here.